document.go

```diff
@@ -206,11 +211,16 @@ func (d Document) Value(field string) (interface{}, bool) {
 
 // SetValue of the field, it returns false if field does not exist, or it's not assignable.
 func (d Document) SetValue(field string, value interface{}) bool {
+	val := d.rv
```

Comment:
my idea was to only non pointer type inside document, so if document is created, d.rv always contains indirect value, not ptr

maybe we should fix the source that cause this issue? maybe I miss some place in collection? or is this related to nested pointer test?

Reply:
removed, this was because of wrong test case